pandas/tests/frame/methods/test_astype.py

```diff
@@ -670,6 +670,20 @@ def test_astype_bytes(self):
         result = DataFrame(["foo", "bar", "baz"]).astype(bytes)
         assert result.dtypes[0] == np.dtype("S3")
 
+    @pytest.mark.parametrize("step1,step2", [(2, 2), (2, 1), (1, 2)])
+    def test_astype_noncontiguous(self, step1, step2):
+        # GH#42396
+        data = np.arange(16).reshape(4, 4)
+        df = DataFrame(data, dtype=np.intp)
+
+        result = df.iloc[:step1, :step2].astype("int16").astype(np.intp)
```

Comment:
using "step" as the parameter name is confusing in this part of the test

Reply:
thanks, have refactored test to clean that up